### PR TITLE
api: add the ability to suspend an organization

### PR DESCRIPTION
**Pull request by mattrobenolt**

### CODE REVIEW

File: src/sentry/models/organization.py [hunk 1]
```diff
@@ -276,3 +277,21 @@ def send_delete_confirmation(self, audit_log_entry, countdown):
             type='org.confirm_delete',
             context=context,
         ).send_async([o.email for o in owners])
+
+    def suspend(self, **kwargs):
+        assert not self.flags.suspended
+
+        self.flags.suspended = True
```
Comment (dcramer) [inline src/sentry/models/organization.py]: would be better if we did an ``update()`` here

File: src/sentry/models/organization.py [hunk 2]
```diff
@@ -276,3 +277,21 @@ def send_delete_confirmation(self, audit_log_entry, countdown):
...
+        self.save()
+
+        if 'owner_id' not in kwargs:
+            kwargs['owner_id'] = self.get_default_owner().id
+
+        if 'owner_email' not in kwargs:
```
Comment (dcramer) [inline src/sentry/models/organization.py]: this is a bit odd, as if you pass owner_id but not owner_email you'll get a mismatch

I'd probably just have them pass owner, and do the rest from within this logic

File: src/sentry/models/suspension.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class Suspension(Model):
+    __core__ = False
+
+    organization_id = BoundedPositiveIntegerField(null=True, unique=True)
+    owner_id = BoundedPositiveIntegerField(null=True)
+    owner_email = models.TextField(null=True, blank=True)
+    reason = models.TextField(null=True, blank=True)
+    added = models.DateTimeField(default=timezone.now)
```
Comment (dcramer) [inline src/sentry/models/suspension.py]: use ``date_added`` to keep things semi understandable based on the rest of the models

**Review by dcramer — CHANGES_REQUESTED** [review]

File: src/sentry/models/suspension.py (hunk 3 above)
Comment (mattrobenolt) [inline src/sentry/models/suspension.py]: Will change.

**Comment by dcramer** [comment]
Looks like this is not valid anymore, as at some point we added 'suspended' and 'suspension_reason' to the schema (along with basic implementations that bubble up to the user).

**Pull request closed without merging**